src/main/scala/com.gu.gibbons/UserReminder.scala

```diff
@@ -27,24 +27,27 @@ class UserReminder[F[_] : Monad](settings: Settings, email: EmailService[F], bon
       * 3- Sends a reminder email to each user 
       * 4- Update keys to log when a reminder has been sent
       */
-    def run(now: Instant, dryRun: Boolean): F[Result] = 
-      if (dryRun) 
-        for {
-          _ <- logger.info("Yop, who's up for receiving a reminder?")
-          users <- bonobo.getUsers(Settings.inactivityPeriod)
-        } yield DryRun(users)
-      else
-        for {
-          _ <- logger.info(s"Getting all the users older than ${Settings.inactivityPeriod}")
-          users <- bonobo.getUsers(Settings.inactivityPeriod)
-          _ <- logger.info(s"Found ${users.length} users. Let's send some emails...")
-          ress <- users.filterNot(u => Settings.whitelist(u.id.id)).traverse { user => 
-            for {
-              newUser <- bonobo.setRemindedOn(user, now)
-              res <- email.sendReminder(newUser)
-            } yield (newUser.id -> res)
-          }.map(_.toMap)
-          _ <- logger.info("aaaand that's a wrap! See you next time.")
-        } yield FullRun(ress)
+    def run(now: OffsetDateTime, dryRun: Boolean): F[Map[UserId, EmailResult]] = {
+      val nowL = now.toInstant.toEpochMilli
+      val thenL = now.minus(Settings.inactivityPeriod).toInstant.toEpochMilli
+      for {
+        _ <- logger.info(s"Getting all developer keys")
+        keys <- bonobo.getDevelopers
```

Comment:
That will _always_ return many thousands of records 😧

Reply:
Cue: we're using DynamoDB as a relational DB 😞

Reply:
Why not query the `User` table by date first, then join to `Key` and filter on tier?

Reply:
💯

Reply:
That is so simple I did not think about it. Closing this PR.